### PR TITLE
Load settings and routes first (bug 1138690)

### PR DESCRIPTION
**Pull request by mstriemer**
Load settings and routes in core/init.js. Depends on https://github.com/mozilla/marketplace-gulp/pull/11 to require `core/init` instead of `main` in the built JS.

### CODE REVIEW

File: core/storage.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ define('core/storage',
                     e.name == 'NS_ERROR_DOM_QUOTA_REACHED') {
                     console.log('LocalStorage full, clearing and reloading');
                     ls.clear();
-                    require('views').reload();
+                    require('core/views').reload();
```
Comment (ngokevin) [inline core/storage.js]: thanks

File: core/init.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+            'site-config': '/api/v2/services/config/site/?serializer=commonplace',
+        });
+
+        router.api.addProcessor(function(endpoint) {
+            return {lang: utils.lang()};
+        });
 
-    router.api.addProcessor(function(endpoint) {
-        return {lang: utils.lang()};
+        require([settings.init_module], function() {});
```
Comment (ngokevin) [inline core/init.js]: I think we can hardcode this as ```main```

Comment (mstriemer) [inline core/init.js]: Since it was hardcoded before you're probably right. I'll change it.

File: core/init.js [hunk 3]
```diff
@@ -1,21 +1,27 @@
 define('core/init',
-    ['core/helpers', 'core/navigation', 'core/polyfill', 'core/router',
-     'core/utils'],
-    function(helpers, navigation, polyfill, router,
-             utils) {
+    ['settings_app', 'routes'],
```
Comment (ngokevin) [inline core/init.js]: we can alphabetize these right?

File: core/storage.js [hunk 4]
```diff
@@ -31,12 +31,13 @@ define('core/storage',
     }
 
     function mapKey(key) {
-        return settings.storage_version + '::' + key;
```
Comment (ngokevin) [inline core/storage.js]: What was wrong with this again? I know we're not using it but don't know if hardcoding it changes much. Although ```settings.storage_version || '0'``` might've been better

**Comment by ngokevin** [comment]
r+wc thanks, i like not having the consumer of mcm not having to handle this

File: core/storage.js (hunk 4 above)
Comment (mstriemer) [inline core/storage.js]: I did not mean to commit this. I'll remove it.

**Comment by mstriemer** [comment]
Closing in favour of https://github.com/mozilla/fireplace/pull/1054.

**Pull request closed without merging**